packages/grafana-data/src/field/fieldDisplay.ts

```diff
@@ -152,6 +152,13 @@ export const getFieldDisplayValues = (options: GetFieldDisplayValuesOptions): Fi
             }
           }
 
+          // Palette color modes use series index (field index) which does not work for when displaing rows
+          // So updating seriesIndex here makes the palette color modes work in "All values" mode
+          field.state = {
```

Comment:
Nit: would it be worth to change this to something like `field.state = setIndexForPaletteColor(field.state)` and move the code + comment into a small help function.

Reply:
:+1: